include/boost/gil/image.hpp

```diff
@@ -101,6 +101,19 @@ class image {
        allocate_and_copy(img.dimensions(),img._view);
     }
 
+    // TODO Optimization: use noexcept (requires _view to be nothrow copy constructible)
+    image(image&& img) :
+      _view(img._view),
+      _memory(img._memory),
+      _align_in_bytes(img._align_in_bytes),
+      _alloc(std::move(img._alloc)),
+      _allocated_bytes(img._allocated_bytes)  {
+        img._view = view_t();
+        img._memory = nullptr;
+        img._align_in_bytes = 0;
+        img._allocated_bytes = 0;
```

Comment:
I wish we had `std::exchange` in C++11 :)

Reply:
Any chance that there is a substitute in Boost.Core somewhere?

Reply:
Ha, a brilliantly obvious question: https://github.com/boostorg/core/blob/develop/include/boost/core/exchange.hpp